xarray/backends/zarr.py

```diff
@@ -363,41 +376,30 @@ def prepare_variable(self, name, variable, check_encoding=False,
         dtype = variable.dtype
         shape = variable.shape
 
-        # TODO: figure out how zarr should deal with unlimited dimensions
-        self.set_necessary_dimensions(variable, unlimited_dims=unlimited_dims)
-
         fill_value = _ensure_valid_fill_value(attrs.pop('_FillValue', None),
                                               dtype)
 
-        # TODO: figure out what encoding is needed for zarr
         encoding = _extract_zarr_variable_encoding(
             variable, raise_on_invalid=check_encoding)
 
-        # arguments for zarr.create:
-        # zarr.creation.create(shape, chunks=None, dtype=None,
-        # compressor='default', fill_value=0, order='C', store=None,
-        # synchronizer=None, overwrite=False, path=None, chunk_store=None,
-        # filters=None, cache_metadata=True, **kwargs)
+        encoded_attrs = OrderedDict()
+        # the magic for storing the hidden dimension data
+        encoded_attrs[_DIMENSION_KEY] = dims
+        for k, v in iteritems(attrs):
+            encoded_attrs[k] = _encode_zarr_attr_value(v)
+
         if name in self.ds:
             zarr_array = self.ds[name]
+            zarr_array.attrs.update(encoded_attrs)
```

Comment:
I'm trying to understand when we would ever hit this case. It seems like it would only happen if we were trying to modify an existing store in place. That is not currently supported.